### DISCUSSION TITLE
Output On Connect or Get current value option for Helper Nodes

### DISCUSSION THREAD

**Discussion by legomind** [zachowj/node-red-contrib-home-assistant-websocket#1226] [Feature Request]
I have several text and switch nodes working to output their states upon changing in ha, but there seems to be no way to get their value on demand or upon connect. Am I missing something?

My use case is an automation to sync the user codes to all of my zwave deadbolts. I have a text node and a switch node for each slot. My automation sets or clears codes to all of the locks accordingly. The problem is that if I restart node red and turn the enable switch on, since the code has not changed, my flow has no way to acquire the value of the text sensor.

This would  be solved if the nodes were to output their current state upon connection to home assistant, or provide a way to fetch the current value on demand.

**Comment by zachowj** [comment]
You can use the current state node to get the current state of any entity. The text node has an option to output the current value.

**Reply by legomind** [reply]
So i would have to do this by device id instead of using a configuration node... Not ideal but i guess that works.

**Reply by zachowj** [reply]
I am not sure I fully follow what you're trying to do.

**Reply by legomind** [reply]
Here is a screenshot of my flow. I am trying to avoid having to use the highlighted nodes.

The Schlage Deadbolt Security subflow must be initialized with two messages one containing the code and one containing the enabled/disabled status, so that when either the code or the enabled disabled switch are changed, the subflow has the latest of each sensor so it can call the set_lock_usercode or clear_lock_usercode services accordingly.

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/3710857/561ddf18-b83a-4489-8b7a-2374b48f4585)

**Comment by DerOetzi** [comment]
+1

Have same construct like @legomind  if you need a further example i can post it

**Reply by DerOetzi** [reply]
Now I'm at home and could prepare an example. My flow looks like this:

![image](https://github.com/user-attachments/assets/c59158ff-e2f2-4fe5-ba1a-f170e762c917)

the two current_state nodes, which are marked, are only triggered once on startup to get the state of the two entity nodes above. They wouldn't be neccessary if the entity nodes would have an "Output on connect" option as for example the event state or trigger state nodes have.

**Reply by DerOetzi** [reply]
Any progress on this?
